cmake/GNU.cmake

```diff
@@ -1,24 +1,23 @@
 
 set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -ffree-line-length-none -fno-range-check -fbacktrace")
 
+
 if(DEBUG)
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Wall -O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check")
```

Comment:
This is not the correct way of adding `DEBUG` flags.
Please add DEBUG flags in CMAKE variable for e.g.
```
set(CMAKE_Fortran_FLAGS_DEBUG "-Wall -O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check" )
```
Cmake will automatically add these flags when the build_type is DEBUG.
See 
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html#variable:CMAKE_BUILD_TYPE

Please do this where `DEBUG` flags are set.